src/fragments/Graph.react.js

```diff
@@ -154,11 +207,62 @@ class PlotlyGraph extends Component {
         clearExtendData();
     }
 
-    graphResize() {
+    getConfig(config, responsive) {
+        return mergeDeepRight(
+            config,
+            responsive ? RESPONSIVE_CONFIG : UNRESPONSIVE_CONFIG
+        );
+    }
+
+    getLayout(layout, responsive) {
+        if (!layout) {
+            return layout;
+        }
+
+        return mergeDeepRight(
+            layout,
+            responsive ? RESPONSIVE_LAYOUT : UNRESPONSIVE_LAYOUT
+        );
+    }
+
+    isResponsive(props) {
+        const {config, figure, responsive} = props;
+
+        if (type(responsive) === 'Boolean') {
+            return responsive;
+        }
+
+        return (
+            (config.responsive || isNil(config.responsive)) &&
+            (!figure.layout ||
+                ((figure.layout.autosize || isNil(figure.layout.autosize)) &&
+                    (isNil(figure.layout.height) ||
+                        isNil(figure.layout.width))))
+        );
+    }
+
+    async graphResize(force = false) {
+        if (!force && !this.isResponsive(this.props)) {
+            return;
+        }
+
         const gd = this.gd.current;
-        if (gd) {
-            Plotly.Plots.resize(gd);
+        if (!gd) {
+            return;
         }
+
+        ++this._resizes;
+        gd.classList.add('dash-graph-resizing');
+
+        Plotly.Plots.resize(gd)
+            .catch(() => {})
+            .finally(() => {
+                --this._resizes;
```

Comment:
We don't really care if it's successful or not, calls and responses just need to balance out to know we're "ready"